### PR TITLE
Add code for python visualization service

### PR DESCRIPTION
**Pull request by ajchili**
This PR adds code to create a visualization service for Kubeflow Pipelines. Currently, it lacks deployment and service yaml files, these will be added with an additional PR once necessary. Additionally, due to python 3 issues encountered with Bazel, the tests are not run automatically.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1651)

### CODE REVIEW

**Comment by ajchili** [comment]
/hold
Adding additional comments and explanations.

**Comment by ajchili** [comment]
/hold cancel
Additional comments and explanations have been added.

**Comment by ajchili** [comment]
/assign @Ark-kun 
bump - readability review

File: backend/Dockerfile.visualization [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2019 Google LLC
```
Comment (kevinbache) [inline backend/Dockerfile.visualization]: maybe include a quick readme which has a two line synopsis of the overall flow.  dockerfile starts server.py which accepts a post request which ... 
makes it easy for the reader to zoom in on the most important parts in the least amount of time.

File: backend/src/apiserver/visualization/exporter.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+# utilizes the basic.tpl file for visualizations then wraps that output with
+# additional tags for javascript and style support. This is ideal for generating
+# visualizations that will be displayed via the frontend.
+class TemplateType(Enum):
+    basic = 'basic'
+    full = 'full'
+
+
+class Exporter:
+
```
[the remark this answers is not in the document]
Comment (ajchili) [reply backend/src/apiserver/visualization/exporter.py]: Did not add due to PEP8 complaining about additional whitespace.

File: backend/src/apiserver/visualization/exporter.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    # Returns the generated HTML as a string.
+    def generate_html_from_notebook(
+            self,
+            nb: NotebookNode,
+            template_type: TemplateType = TemplateType.full
+    ) -> Text:
+        # HTML generator and exporter object
+        html_exporter = HTMLExporter()
+        template_file = "templates/{}.tpl".format(template_type.value)
+        html_exporter.template_file = str(Path.cwd() / template_file)
```
[the remark this answers is not in the document]
Comment (ajchili) [reply backend/src/apiserver/visualization/exporter.py]: That would require the usage of [`PurePath.parent`](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parent) to determine the parent route of the file, is that ideal for this?

**Comment by ajchili** [comment]
/hold
Recent changes have resulted in unexpected latency for generated visualizations. I am investigating why this is occurring.

**Pull request merged**